src/lib/marlin_plonk_bindings/js/bindings.js

```diff
@@ -736,3 +825,1210 @@ var caml_pasta_vesta_endo_scalar = plonk_wasm.caml_pasta_vesta_endo_scalar;
 var caml_pasta_vesta_affine_deep_copy = function(pt) {
     return rust_affine_to_caml_affine(plonk_wasm.caml_pasta_vesta_affine_deep_copy(rust_affine_of_caml_affine(pt, plonk_wasm.caml_pasta_vesta_affine_one)));
 };
+
+
+
+
+// Provides: caml_array_of_rust_vector
```

Comment:
Does all this code need to be in JS? It almost seems like it should be in OCaml and compiled to JS.